### ISSUE TITLE
RestSharp 106.11.7 In some cases ExecuteAsync with Method POST never respond

### ISSUE THREAD

**Issue by luke92** [restsharp/RestSharp#1535]
Hi all!
I'm using RestSharp 106.11.7 and i have problems with an URL with ExecuteAsync using Method.POST
The method never respond (I try debug and not debug).
For other methods it's OK.
Also i use POSTMAN and the POST it's OK.

This is my code (Works with other URL with Method GET and POST)

`var urlMethod = "/v1/receipt";
                var client = GetClient();
                var request = await GetRequest(urlMethod, Method.POST, JsonConvert.SerializeObject(recibo), true);
                var response = await client.ExecuteAsync(request);`

`private RestClient GetClient()
        {
            var baseUrl = GetLoginCredentials().Endpoint;
            var client = new RestClient(baseUrl);
            client.Timeout = -1;
            return client;
        }`

`private async Task<RestRequest> GetRequest(string urlMethod, Method method, string json, bool needToken)
        {
            if (needToken)
            {
                var tokenRenew = await RenewToken();
                request.AddHeader("Authorization", "Bearer " + _token.Access_token);
            }
            var request = new RestRequest(urlMethod, method);
            request.AddHeader("Content-Type", "application/json");
            if (!string.IsNullOrEmpty(json))
                request.AddParameter("application/json", json, ParameterType.RequestBody);
            return request;
        }`

**Comment by mhornbacher** [comment]
Your code is a bit hard to follow formatted like that. Please correct me if this is wrong:

```csharp
var urlMethod = "/v1/receipt";
var client = GetClient();
var request = await GetRequest(urlMethod, Method.POST, JsonConvert.SerializeObject(recibo), true);
var response = await client.ExecuteAsync(request);

private RestClient GetClient() {
    var baseUrl = GetLoginCredentials().Endpoint;
    var client = new RestClient(baseUrl);
    client.Timeout = -1;
    return client;
}

private async Task<RestRequest> GetRequest(string urlMethod, Method method, string json, bool needToken) {
    if (needToken) {
        var tokenRenew = await RenewToken();
        request.AddHeader("Authorization", "Bearer " + _token.Access_token);
    }
    var request = new RestRequest(urlMethod, method);
    request.AddHeader("Content-Type", "application/json");
    if (!string.IsNullOrEmpty(json))
        request.AddParameter("application/json", json, ParameterType.RequestBody);
    return request;
}
```

I'm not sure how this compiles, as `request` is used in `GetRequest` before it is assigned. Also `tokenRenew` does not seem to be used and seems to be updating the `_token` field. Is this correct?

Is your problem (in the debugger) that it hangs indefinitely on `client.ExecuteAsync(request)`? If so did you:

1. Try setting the timeout, to see if that is an issue
2. Try logging the final request to ensure that this is what you are sending from postman? (perhaps there are some additional headers or the url/protocol is slightly different)

**Comment by luke92** [comment]
Hi @mhornbacher  how are you?

- I create a new request with the Bearer token in the header.
- RenewToken update _token if it's necessary 
- Yes, it's hang in debugger but also without debug.

1. Default timeout is -1 but i configure a timeout with 15 seconds and it's the same problem.
2. I think it's the same header as Postman

**Comment by mhornbacher** [comment]
I think `-1` is an infinite timeout. If you remove that the request should just eventually fail.

Are you able to export your postman request?

Sadly we cannot see the rest of your code, but I would take a strong look at the final URL. You can see this with `client.BuildUri(request)`
